src/sidebar/build-thread.js

```diff
@@ -305,86 +297,99 @@ const defaultOpts = {
  * the current visibility filters and sort function and returns
  * the thread structure that should be rendered.
  *
+ * An Annotation present in `annotations` will not be present in the returned threads if:
+ * - The annotation does not match thread-level filters (options.threadFilterFn), OR
+ * - The annotation is not in the current selection (options.selected), OR
+ * - The annotation's thread is hidden and has no visible children
+ *
+ * Annotations that do not match the currently-applied annotation filters
+ * (options.filterFn) will have their thread's `visible` property set to `hidden`
+ * (an exception is made if that annotation's thead has been forced visible by
+ * a user).
+ *
  * @param {Array<Annotation>} annotations - A list of annotations and replies
```

Comment:
```suggestion
 * @param {Annotation[]} annotations - A list of annotations and replies
```

While you're improving types here, I suggest for consistency we should use `T[]` rather than `Array<T>` except for the cases where `T` is some expression that can't be used with the shorter syntax.

Reply:
Ah! I had just commented about this above! Great minds...